plasma_framework/truffle-config.js

```diff
@@ -31,22 +34,16 @@ module.exports = {
         },
         infura: {
             skipDryRun: true,
-            provider: () => {
-                const infuraUrl = `${process.env.INFURA_URL}/${process.env.INFURA_API_KEY}`;
-
-                // Replace double '//'
-                const cleanInfuraUrl = infuraUrl.replace(/([^:])(\/\/+)/g, '$1/');
-
-                return new HDWalletProvider(
-                    [
-                        process.env.DEPLOYER_PRIVATEKEY,
-                        process.env.MAINTAINER_PRIVATEKEY,
-                        process.env.AUTHORITY_PRIVATEKEY,
-                    ],
-                    cleanInfuraUrl,
-                    0, 3,
-                );
-            },
+            // Jesus fuck why can't this be a function
+            provider: new HDWalletProvider(
+                [
+                    process.env.DEPLOYER_PRIVATEKEY || '0'.repeat(64),
```

Comment:
I don't really understand, why we would need empty 0s here?

Reply:
It fails on tests if the private keys are unfmdefined since infuras wallet provider is not a function, but rather instanciates a class on runtime. This instantiation reads the keys and trys to derive it but immediately throws an error because the keys are undefined. To circumvent this I've made the default keys to be zeros